### PR TITLE
Add contribution guidelines and issue templates.

### PR DESCRIPTION
**Pull request by FtZPetruska**
## Description

This PR adds contributing guidelines to the project as well as two issue templates (one for bug reports, the other for enhancements).

The issue templates can be seen [here](https://github.com/FtZPetruska/SDL/issues/new/choose), and the contributing file [here](https://github.com/FtZPetruska/SDL/blob/main/docs/CONTRIBUTING.md).

Let me know what you think about it and if you wish for any change to be made!

## Existing Issue

#5610

### CODE REVIEW

**Comment by slouken** [comment]
I'm kind of inclined not to have the bug templates. We get a lot of bugs from users who can't for whatever reason create a small repro case and many of them have special case circumstances that don't fit neatly into a template. I think we do a pretty good job with the bug reports we get and there isn't a lot of value by adding them. @icculus, what are your thoughts?

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name: Report a bug
 about: Let us know about a bug.
 title: ""
-labels: ""
+labels: "bug"
```
Comment (slouken) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:5]: If we do accept the templates, please don't add the bug label. Almost everything is a bug and this just adds noise.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    -   [Editing the wiki](#editing-the-wiki)
+
+## Filing a GitHub issue
+
+### Reporting a bug
+
+If you think you have found a bug and would like to report it, here are the steps you should take:
+
+-   Before opening a new issue, ensure your bug has not already been reported on the [GitHub Issues page](https://github.com/libsdl-org/SDL/issues).
+-   Prepare a small example that reproduces your bug.
```
Comment (slouken) [inline docs/CONTRIBUTING.md:26]: "bug." -> "bug, if possible."

**Comment by icculus** [comment]
> I'm kind of inclined not to have the bug templates. We get a lot of bugs from users who can't for whatever reason create a small repro case and many of them have special case circumstances that don't fit neatly into a template.

Agreed, and fwiw, when I report a bug in another project and it has a template of any complexity, I'm inclined to close the tab without reporting. We want the lowest barriers to entry for this, even if we have to ask for more information later.

I _do_ think a template could be useful, less for structure and detail, if there was a "hey, thanks for reporting a bug...it's REALLY HELPFUL if you tell us upfront (this one important thing that keeps coming up)" ... but at the moment I can't think of something important enough to do that.

**Comment by eloj** [comment]
I'm the same with overly long templates. Don't particularly like them as a user.

I don't mind if the Issue template contains a "(please state your platform and SDL version") and the PR template contains a "Fixes #nnnn" as a reminder, but that's about it. I don't see how you'd almost ever need more than two or three lines of text in one of these.

**Comment by slouken** [comment]
@FtZPetruska, can you simplify or remove the templates and update the contributing documentation accordingly?

**Comment by FtZPetruska** [comment]
Thank you everyone for your feedback!

I removed issue templates were removed in favour of adding short instructions in the guidelines:
```
-   Include details about your environment, such as your Operating System and SDL version.
-   If possible, provide a small example that reproduces your bug.
```

**Comment by slouken** [comment]
Looks good, thanks!

**Pull request merged**